pytorch_lightning/strategies/ipu.py

```diff
@@ -134,6 +135,11 @@ def setup(self, trainer: "pl.Trainer") -> None:
 
         super().setup(trainer)
 
+        # disable the `optimizer_zero_grad` function by setting it to `None`.
+        # this is because the IPU zeros the gradients internally
+        self._optimizer_zero_grad_original = self.lightning_module.optimizer_zero_grad
```

Comment:
You need to define this in the `__init__` too, as done for `_update_dataloader_original`

Should resolve the current IPU failures